contracts/JoeBarV2.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+interface IBarRewarder {
+    function claimReward() external;
+}
+
+// JoeBar is the coolest bar in town. You come in with some Joe, and leave with more! The longer you stay, the more Joe you get.
+//
+// This contract handles swapping to and from moJoe, JoeSwap's staking token.
+contract JoeBarV2 is Initializable, ERC20Upgradeable, OwnableUpgradeable {
+    using SafeMathUpgradeable for uint256;
+
+    IERC20Upgradeable public joe;
+    IBarRewarder public rewarder;
+    uint256 public entryFee;
+
+    event SetEntryFee(uint256 _entryFee);
+    event SetRewarder(address _rewarder);
+
+    // Define the Joe token contract
+    function initialize(IERC20Upgradeable _joe, uint256 _entryFee) public initializer {
+        __ERC20_init("JoeBarV2", "moJOE");
+        __Ownable_init();
+
+        joe = _joe;
+        setEntryFee(_entryFee);
+    }
```

Comment:
Should we initialize rewarder too here?

Reply:
Issue is rewarder initialize bar, so we can't initialize bar with rewarder too (it loops back)

Reply:
In this case, I would add an `initialized` boolean variable and `initialize()` function (in addition to the constructor) in `BarRewarderPerSec` and then in the JoeBarV2 constructor, we pass in the rewarder address and call `rewarder.initialize()`.

So steps should be:

 1. Deploy BarRewarder
 2. Deploy JoeBarV4 -- inside constructor we call `initialize()` on BarRewarder
Similar logic can be found when we set RewarDistributor for the Joetroller:

https://github.com/traderjoe-xyz/joe-lending/blob/main/contracts/Joetroller.sol#L1019-L1032
https://github.com/traderjoe-xyz/joe-lending/blob/main/contracts/RewardDistributor.sol#L84-L94